src/resolver/Resolver.js

```diff
@@ -34,117 +34,120 @@ function DsrResolver() {
+        // the type of the VC eg civ:Type:address or cvc:Identity:name
+        const globalIdentifierType = globalIdentifier.substring(0, globalIdentifier.indexOf('-'));
 
-      // for credentials we filter out the credentials, the meta issuer than the claim path
-      if (globalIdentifierType === 'credential') {
-        const type = globalIdentifier.substring('credential-'.length, globalIdentifier.lastIndexOf('-'));
-        // filter the VCs, do not confuse this $eq with the operator $eq on credentialItems array
-        const tempFiltered = credentials.filter(sift({ identifier: { $regex: `${type}` } }));
+        // for credentials we filter out the credentials, the meta issuer than the claim path
+        if (globalIdentifierType === 'credential') {
+          const type = globalIdentifier.substring('credential-'.length, globalIdentifier.lastIndexOf('-'));
+          // filter the VCs, do not confuse this $eq with the operator $eq on credentialItems array
+          const tempFiltered = credentials.filter(sift({ identifier: { $regex: `${type}` } }));
 
-        // if there is an constraint on the credential
-        if (credentialItem.identifier) {
-          const filterArgArray = [];
-          // filtering meta constraints if they exist
-          if (credentialItem.constraints.meta) {
-            if (credentialItem.constraints.meta.issued) {
-              // there is only one key
-              const operatorIssued = Object.keys(credentialItem.constraints.meta.issued.is)[0];
-              const convertedOperatorIssued = this.convertMongoOperatorToJavascript(Object.keys(credentialItem.constraints.meta.issued.is)[0]);
-              const claimConstraintIssued = credentialItem.constraints.meta.issued.is[operatorIssued];
-              const claimFilterIssued = {};
-              claimFilterIssued.$where = `new Date(this.issued).getTime() ${convertedOperatorIssued} ${claimConstraintIssued}`;
-              filterArgArray.push(claimFilterIssued);
-            }
-            if (credentialItem.constraints.meta.expiry) {
-              // there is only one key
-              const operatorExpiry = Object.keys(credentialItem.constraints.meta.expiry.is)[0];
-              const convertedOperatorExpiry = this.convertMongoOperatorToJavascript(Object.keys(credentialItem.constraints.meta.expiry.is)[0]);
-              const claimConstraintExpiry = credentialItem.constraints.meta.expiry.is[operatorExpiry];
-              const claimFilterExpiry = {};
-              claimFilterExpiry.$where = `new Date(this.expiry).getTime() ${convertedOperatorExpiry} ${claimConstraintExpiry}`;
-              filterArgArray.push(claimFilterExpiry);
+          // if there is an constraint on the credential
+          if (credentialItem.identifier) {
+            const filterArgArray = [];
+            // filtering meta constraints if they exist
+            if (credentialItem.constraints.meta) {
+              if (credentialItem.constraints.meta.issued) {
```

Comment:
it seems like there's some code duplication in these conditionals. How about a function like this: 
`
const parseClaimFilter = (claimConstraint) => {
	const operatorIssued = Object.keys(claimConstraint.is)[0];
	const convertedOperatorIssued = this.convertMongoOperatorToJavascript(Object.keys(claimConstraint.is)[0]);
	const claimConstraintIssued = claimConstraint.is[operatorIssued];
	return { operator, constraint };
}
`
then this is called for each constraint:
`
const { operator: convertedOperatorIssued, constraint: claimConstraintIssued } = parseClaimFilter(credentialItem.constraints.meta.issued);
const claimFilterIssued = {};
...
`

Reply:
Hmm, I like the old code better :P, will ignore this one

Reply:
fair enough